libfreerdp/core/gateway/http.c

```diff
@@ -454,6 +454,7 @@ static BOOL http_response_parse_header_status_line(HttpResponse* response, char*
 		if ((errno != 0) || (val < 0) || (val > INT16_MAX))
 			return FALSE;
 
+		errno = 0;
 		response->StatusCode = strtol(status_code, NULL, 0);
```

Comment:
It should be "response->StatusCode = val;" instead, "status_code" was already parsed into "val" on line 452.

Reply:
Right, I didn't apply any thought to whether the context made any sense- this was kind of a low priority for me as we're preparing for an upcoming release over in our world. =) ETIME, as it goes. =(